test/runtests.jl

```diff
@@ -43,6 +44,19 @@ end
         @test occursin("first", logtext)
         @test occursin("second", logtext)
         @test occursin("third", logtext)
+        # IOStream
+        filepath = joinpath(dir, "log2")
+        open(filepath, "w") do filehandle
+            with_logger(FileLogger(filehandle)) do
+                @info "first"
+                @warn "second"
+                @info "third"
+            end
+        end
+        logtext = String(read(filepath))
+        @test occursin("first", logtext)
+        @test occursin("second", logtext)
+        @test occursin("third", logtext)
     end
```

Comment:
Can we write this to use a nested 
```julia
@testset "$(typeof(file_obj))" for file_obj in (filepath, open(filepath*"_io", w))
```

and thus avoid dublicating code

Reply:
I had something like that before but ideally we want to test reading from the file and not the stream. Also, it might be better to properly close the file before reading to make sure it is flushed properly (?). Let me try to reduce the code though.